librz/socket/run.c

```diff
@@ -32,6 +32,17 @@
 #include <mach-o/nlist.h>
 #endif
 
+#if HAVE_OPENPTY && HAVE_FORKPTY && HAVE_LOGIN_TTY
+#if defined(__APPLE__) || defined(__NetBSD__) || defined(__OpenBSD__)
+#include <util.h>
+#elif defined(__FreeBSD__) || defined(__DragonFly__)
+#include <libutil.h>
+#else
+#include <pty.h>
+#include <utmp.h>
+#endif
+#endif
```

Comment:
In theory, also these should be mesonified :D `meson.has_header`. There's a lot of improvements that can be done by using build system checks instead of manually defined things. In the moment you want to add support for a new system you don't have to do anything (not in this regard at least...)